File: laramie-tesoreria-webapi/src/domain/entities/pago-rendicion-lote.ts
export default class PagoRendicionLote {

    id: number;
	numeroLote: string;
	fechaLote: Date;
	casos: number;
	importeTotal: number;
	idUsuarioProceso: number;
	fechaProceso: Date;
	fechaConfirmacion: Date;

	constructor(
        id: number = 0,
		numeroLote: string = "",
		fechaLote: Date = null,
		casos: number = 0,
		importeTotal: number = 0,
		idUsuarioProceso: number = 0,
		fechaProceso: Date = null,
		fechaConfirmacion: Date = null
	)
	{
        this.id = id;
		this.numeroLote = numeroLote;
		this.fechaLote = fechaLote;
		this.casos = casos;
		this.importeTotal = importeTotal;
		this.idUsuarioProceso = idUsuarioProceso;
		this.fechaProceso = fechaProceso;
		this.fechaConfirmacion = fechaConfirmacion;
	}

	setFromObject = (row) =>
	{
        this.id = row.id ?? 0;
		this.numeroLote = row.numeroLote ?? "";
		this.fechaLote = row.fechaLote ?? null;
		this.casos = row.casos ?? 0;
		this.importeTotal = row.importeTotal ?? 0;
		this.idUsuarioProceso = row.idUsuarioProceso ?? 0;
		this.fechaProceso = row.fechaProceso ?? null;
		this.fechaConfirmacion = row.fechaConfirmacion ?? null;
	}

}

File: modules/tesoreria/models/pago_rendicion_lote.py
from sqlalchemy import Column, BigInteger, String, Integer, Date, DateTime, Numeric
from sqlalchemy.orm import relationship

import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from shared.database import Base


class PagoRendicionLote(Base):
    __tablename__ = "tesoreria_pago_rendicion_lotes"

    id = Column(BigInteger, primary_key=True)
    numero_lote = Column(Integer, nullable=False)
    fecha_lote = Column(Date, nullable=False)
    casos = Column(Integer, nullable=False, default=0)
    importe_total = Column(Numeric(18, 2), nullable=False, default=0)
    id_usuario_proceso = Column(BigInteger, nullable=True)
    fecha_proceso = Column(DateTime, nullable=True)
    fecha_confirmacion = Column(DateTime, nullable=True)

    rendiciones = relationship("PagoRendicion", back_populates="lote", lazy="dynamic")
